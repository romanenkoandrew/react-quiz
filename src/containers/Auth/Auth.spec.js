import { shallow } from "enzyme";
import { Auth } from "./Auth";

describe("Auth component", () => {
  let auth, component, instance;
  beforeEach(() => {
    auth = jest.fn();
    component = shallow(<Auth auth={auth} />);
    instance = component.instance();
  });
  it("Auth component", () => {
    expect(component).toMatchSnapshot();
  });
  it("should call method submitHandler", () => {
    const e = { preventDefault: jest.fn() };
    instance.submitHandler(e);
    expect(e.preventDefault).toBeCalledTimes(1);
  });
  it("should call auth in method loginHandler", () => {
    instance.loginHandler();
    expect(auth).toBeCalledTimes(1);
  });
  it("should call auth in method registerHandler", () => {
    instance.registerHandler();
    expect(auth).toBeCalledTimes(1);
  });
  it("should call the validateControl method without validation", () => {
    expect(instance.validateControl("value")).toBe(true);
  });
  it("should call the validateControl method with validation", () => {
    expect(
      instance.validateControl("test@mail.ru", {
        required: true,
        minLength: 5,
        email: "test@mail.ru",
      })
    ).toBe(true);
  });
  it("should call the validateControl method with validation and empty value", () => {
    expect(
      instance.validateControl("", {
        required: false,
        minLength: 0,
        email: "",
      })
    ).toBe(true);
  });
  it("should call method onChangeHandler in Auth", () => {
    const e = { target: { value: 1 } };
    instance.setState({ formControls: { controlName: {} } });
    instance.onChangeHandler(e, "controlName");
    expect(instance.state.formControls.controlName.touched).toBe(true);
    expect(instance.state.formControls.controlName.value).toBe(1);
    expect(instance.state.formControls.controlName.valid).toBe(true);
    expect(instance.state.isFormValid).toBe(true);
  });
});
