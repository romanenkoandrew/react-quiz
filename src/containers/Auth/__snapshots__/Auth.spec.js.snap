// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth component Auth component 1`] = `
<div
  className="Auth"
>
  <div>
    <h1>
      Авторизация
    </h1>
    <form
      action=""
      className="AuthForm"
      onSubmit={[Function]}
    >
      <Input
        errorMessage="Введите корректный email"
        key="email0"
        label="Email"
        onChange={[Function]}
        shouldValidate={true}
        touched={false}
        type="email"
        valid={false}
        value=""
      />
      <Input
        errorMessage="Введите корректный пароль"
        key="password1"
        label="Пароль"
        onChange={[Function]}
        shouldValidate={true}
        touched={false}
        type="password"
        valid={false}
        value=""
      />
      <Button
        disabled={true}
        onClick={[Function]}
        type="success"
      >
        Войти
      </Button>
      <Button
        disabled={true}
        onClick={[Function]}
        type="primary"
      >
        Зарегистрироваться
      </Button>
    </form>
  </div>
</div>
`;
